### DISCUSSION TITLE
Backwards compatibility

### DISCUSSION THREAD

**Discussion by AndrolGenhald** [vimeo/psalm#7423] [General]
Can we define somewhere what guarantees we want to provide across minor and major versions? It seems to me that we provide very few guarantees for end-users, but we want to make sure plugins will continue to work until a major version change.

If major versions only matter for plugins, I'm thinking the [installation docs](https://psalm.dev/docs/running_psalm/installation/#installation) should be changed to `composer require --dev vimeo/psalm="*"` so that the latest version of psalm is used based on plugin support, otherwise composer defaults to the current major version which means a lot of users may not automatically upgrade to psalm 5 once it's released.

**Comment by AndrolGenhald** [comment]
I suppose for end users removing annotations between major versions might break things, which might be a good enough reason to default to requiring the current major version.

**Comment by orklah** [comment]
Well, my opinion is that most people should pin Psalm in a specific version to avoid breaking their build on new versions of Psalm (even bugfixes).

Then it should be updated on occasion. 

So yeah, BC break is mainly code related, so it impacts plugins first
